src/dbnode/persist/fs/seek_manager.go

```diff
@@ -774,14 +802,10 @@ func (m *seekerManager) seekersByTime(shard uint32) *seekersByTime {
 	}
 
 	seekersByShardIdx := make([]*seekersByTime, shard+1)
-
-	for i := range seekersByShardIdx {
-		if i < len(m.seekersByShardIdx) {
-			seekersByShardIdx[i] = m.seekersByShardIdx[i]
-			continue
-		}
-		seekersByShardIdx[i] = &seekersByTime{
-			shard:   uint32(i),
+	idx := copy(seekersByShardIdx, m.seekersByShardIdx)
```

Comment:
This was a change I introduced when trying to get some tests to pass (turned out to be unnecessary for that though)

I _think_ it's better but have a look